TransformationSystem/Client/TaskManager.py

```diff
@@ -566,28 +569,37 @@ def __prepareTransformationTasks( self, transBody, taskDict, owner, ownerGroup,
     """ Prepare transformation tasks with a job object per task
     """
     method = '__prepareTransformationTasks'
+    startTime = time.time()
+    oJobTemplate = self.jobClass( transBody )
+    oJobTemplate.setOwner( owner )
+    oJobTemplate.setOwnerGroup( ownerGroup )
+    oJobTemplate.setOwnerDN( ownerDN )
+    site = oJobTemplate.workflow.findParameter( 'Site' ).getValue()
+    jobType = oJobTemplate.workflow.findParameter( 'JobType' ).getValue()
+    templateOK = False
+    outputTime = 0.
     for taskID, paramsDict in taskDict.iteritems():
       # Create a job for each task and add it to the taskDict
-      oJob = self.jobClass( transBody )
-      site = oJob.workflow.findParameter( 'Site' ).getValue()
+      if not templateOK:
+        templateOK = True
+        # Update the template with common information
+        transID = paramsDict['TransformationID']
+        self._logVerbose( 'Job owner:group to %s:%s' % ( owner, ownerGroup ),
+                          transID = transID, method = method )
+        transGroup = str( transID ).zfill( 8 )
+        self._logVerbose( 'Adding default transformation group of %s' % ( transGroup ),
+                          transID = transID, method = method )
+        oJobTemplate.setJobGroup( transGroup )
+        oJobTemplate._setParamValue( 'PRODUCTION_ID', str( transID ).zfill( 8 ) )
+
       paramsDict['Site'] = site
-      jobType = oJob.workflow.findParameter( 'JobType' ).getValue()
       paramsDict['JobType'] = jobType
-      transID = paramsDict['TransformationID']
-      self._logVerbose( 'Setting job owner:group to %s:%s' % ( owner, ownerGroup ),
-                        transID = transID, method = method )
-      oJob.setOwner( owner )
-      oJob.setOwnerGroup( ownerGroup )
-      oJob.setOwnerDN( ownerDN )
-      transGroup = str( transID ).zfill( 8 )
-      self._logVerbose( 'Adding default transformation group of %s' % ( transGroup ),
-                        transID = transID, method = method )
-      oJob.setJobGroup( transGroup )
+      # Now create the job from the template
+      oJob = copy.deepcopy( oJobTemplate )
       constructedName = self._transTaskName( transID, taskID )
       self._logVerbose( 'Setting task name to %s' % constructedName,
                         transID = transID, method = method )
       oJob.setName( constructedName )
-      oJob._setParamValue( 'PRODUCTION_ID', str( transID ).zfill( 8 ) )
       oJob._setParamValue( 'JOB_ID', str( taskID ).zfill( 8 ) )
```

Comment:
So, the only parameter different from 1 job to the other, at this point, is the `JOB_ID`?

It seems correct to me, this comment is just for double-checking.